src/systemsSerology.jl

```diff
@@ -50,10 +50,58 @@ function antigenTables(s::String)
     dfF = CSV.read(joinpath(dataDir, "alter-MSB", "data-function.csv"))
     newdfF = dfF[dfF[!, :ADCC] .!= "NA", :] #ignore subjects who have no ADCC data
     ADCConly = newdfF[:, [:Column1, :ADCC]] #only want ADCC data
+    ADCConly.ADCC = tryparse.(Float64, ADCConly.ADCC) #ADCC data should be Floats not strings
     ADCC = rename!(ADCConly, [:Subject, :ADCC]) #Subjects were called "Column1" before
 
     #join ADCC data with antigen receptor data
     allAntigen = join(rec, ADCC, on = :Subject, kind = :inner)
 
     return allAntigen
 end
+
+""" Assemble the Cube """
+function createCube()
+    dataDir = joinpath(dirname(pathof(FcgR)), "..", "data")
+    dfMA = CSV.read(joinpath(dataDir, "alter-MSB", "meta-antigens.csv"))
+    dfMS = CSV.read(joinpath(dataDir, "alter-MSB", "meta-subjects.csv"))
+    dfMD = CSV.read(joinpath(dataDir, "alter-MSB", "meta-detections.csv"))
+    dfMD.detection = replace.(dfMD.detection, "." => "_")
+    dfMD.Number = 1:22 #Add a column of index numbers that correspond to each receptor/detection
+
+    Cube = Array{Union{Nothing, Float64}}(nothing, 182, 23, 41) #create a the Cube, filled with nothing
+
+    #Subject IDs in column 1 down the cube
+    #Detection number across top of cube - like column names for each slice
+    for i = 2:182
+        for j = 2:23
+            for k = 1:41
+                Cube[i, 1, k] = dfMS.Column1[i-1]
+                Cube[1, j, k] = dfMD.Number[j-1]
+            end
+        end
+    end
+
+    #Massive for loop that will find correct index for each data point in antigen tables and put into correct index in the Cube
+    
+    for p = 1:size(dfMA, 1)
+        A = FcgR.antigenTables(dfMA.antigen[p])    #focus on one antigen at a time (one slice of cube)
+        B = describe(A)                            #want column names in a listed table for later
+        for j = 1:size(dfMD, 1)                    #run through all possible detections/receptors
+            for i in 1:size(B, 1)                  
+                if (Symbol(dfMD.detection[j]) == B.variable[i])    #see if current receptor binds to current antigen (exists in table)
+                    df2 = select(A, Symbol(dfMD.detection[j]), :Subject)  #create a subsetted dataframe for this receptor antigen combo
+                    rename!(df2, [:detection, :Subject])           #need uniform names 
```

Comment:
Couldn't you just have `df.Subject` instead of `df.detection` on line 97?

Reply:
df.Subject is the subject ID (e.g. 753551). df.detection is the column that contains the actual value corresponding to the receptor/antigen. I renamed the column to detection because alter-MSB/meta-detections is the file that contains all the receptor or "detection" names (FcgRI etc). Each sub-dataframe will originally have a specific detection column name, and I wanted a generic column name "detection" to call each loop. So here I'm finding the correct row in the dataframe (rows indexed by subjects) and then moving over a column to find that row's value.